SwiftCompilerSources/Sources/Optimizer/Utilities/EscapeInfoVisitor.swift

```diff
@@ -0,0 +1,576 @@
+import SIL
+
+struct EscapeInfoState : HasResult {
+  let result: WalkResult
+  let followStores: Bool
+  let knownType: Type?
+  let walkDown: Bool
+  
+  init(result: WalkResult, followStores: Bool, knownType: Type?, walkDown: Bool = false) {
+    self.result = result
+    self.followStores = followStores
+    self.knownType = knownType
+    self.walkDown = walkDown
+  }
+  
+  func with(result: WalkResult) -> EscapeInfoState {
+    return EscapeInfoState(result: result, followStores: followStores, knownType: knownType)
+  }
+  
+  func with(knownType: Type?) -> EscapeInfoState {
+    return EscapeInfoState(result: result, followStores: followStores, knownType: knownType)
+  }
+  
+  func with(followStores: Bool) -> EscapeInfoState {
+    return EscapeInfoState(result: result, followStores: followStores, knownType: knownType)
+  }
+  
+  func with(walkDown: Bool) -> EscapeInfoState {
+    return EscapeInfoState(result: result, followStores: followStores, knownType: knownType, walkDown: walkDown)
+  }
+}
+
+protocol VisitUseFunction : HasState {
+  mutating
+  func visitUse(operand: Operand, path: SmallProjectionPath, state: State) -> State
```

Comment:
Do we really need to pass a `state` to EscapeInfo's visitUse and visitDef functions?